test/test-users.js

```diff
@@ -0,0 +1,402 @@
+const chai = require('chai');
+const chaiHttp = require('chai-http');
+const spies = require('chai-spies');
+const mongoose = require('mongoose');
+const app = require('../server').app;
+
+const {
+  User
+} = require('../users/models');
+const should = chai.should();
+
+var UrlPattern = require('url-pattern');
+const pattern = {
+  "/users": new UrlPattern('/users'),
+  "/users/:userId": new UrlPattern('/users/:userId')
+}
+
+chai.use(chaiHttp);
+chai.use(spies);
+
+describe('Test /user endpoints with empty database', () => {
+  beforeEach((done) => {
+    // Approach #1 Drop database and `ensureIndexes()`
+    mongoose.connection.db.dropDatabase()
```

Comment:
No objection to `mongoose.connection.dropDatabase();`. Do you prefer the dropDatabase and ensureIndexes over the and User.remove({})??